presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoConnection.java

```diff
@@ -125,6 +125,18 @@ public class PrestoConnection
         initializeQueryInterceptors();
     }
 
+    PrestoConnection(PrestoConnection connection, PrestoDriverUri uri, QueryExecutor queryExecutor)
```

Comment:
It's not obvious to me that all this does is copy over the session properties.  Why not extra credentials or custom headers for example?  A couple of options that might make it more obvious:

1) Make a `ConnectionBuilder`, where you can specify the things you want to override.
2) Just make a simple static method that's called something explicit, like `PrestoConnection.newConnectionWithSessionProperties`, and that takes in a map of the session properties.

Reply:
I want to reuse existing open connection session properties to create a new connection.  PrestoConnection getSessionProperties is not public (if it were, none of this would be required) 

The use case is as follows:
- A connection is open to execute a query (By Tableau in my use case)
- The query is intercepted via PrestoQueryInterceptor
- Query is evaluated
- A new connection is created (removing the query interceptor property and injecting other properties)
    -  A new connection is required to remove the query interceptor property and prevent an interceptor infinite loop
    - Once a new connection is created all previously configured connection session properties get lost. There's currently no way to get those properties from a connection instance due to the visibility of the getSessionProperties in PrestoConnection class
 
Suggestion 2 (PrestoConnection.newConnectionWithSessionProperties) sounds good to me, but I guess I will need to pass a PrestoConnection parameter unless there's a way to get the sessions properties from a connection instance (I think there is not)

Does that make sense?

Reply:
Makes sense to me